flask_common/documents.py

```diff
@@ -438,3 +438,18 @@ def _get_query_shape(self, query):
                 query_shape[key] = 1
         return query_shape
 
+
+def iter_no_cache(query_set):
+    """Iterate over queryset without caching it.
+
+    Useful for iterating over large result sets / bulk actions.
+
+    If a batch size is not set, apply a sensible default of 1000
+    that's better than what Mongo server is doing (101 first and
+    then as many as it can fit in 4MB) to avoid cursor timeouts.
+    """
+    if query_set._batch_size is None:
```

Comment:
@thomasst I also added a default batch size in case anyone forgets an explicit one.